src/Serializer.php

```diff
@@ -118,8 +133,172 @@ public static function decodeMetadata($metadata)
         return $result;
     }
 
+    private function encodeElement(FieldDescriptor $field, $data)
```

Comment:
Why do we need to do all of this encoding/decoding work - why doesn't protobuf have this already?